polling/poller.go

```diff
@@ -32,13 +37,33 @@ func NewLongRunningPollerFromResponse(ctx context.Context, resp *http.Response,
 	return poller, nil
 }
 
+// NewPollerFromResponse creates a new LongRunningPoller from the HTTP Response
+func NewPollerFromResponse(ctx context.Context, resp *http.Response, client autorest.Client, method string) (LongRunningPoller, error) {
+	poller := LongRunningPoller{
+		ctx:    ctx,
+		client: client,
+		method: method,
+	}
+	future, err := azure.NewFutureFromResponse(resp)
+	if err != nil {
+		return poller, err
+	}
+	poller.future = &future
+	return poller, nil
+}
+
 // PollUntilDone polls until this Long Running Poller is completed
 func (fw *LongRunningPoller) PollUntilDone() error {
 	if fw.future == nil {
 		return fmt.Errorf("internal error: cannot poll on a nil-future")
 	}
 
 	err := fw.future.WaitForCompletionRef(fw.ctx, fw.client)
+	if strings.EqualFold(fw.method, "DELETE") {
+		if !response.WasNotFound(fw.future.Response()) {
+			return err
```

Comment:
since we're returning `err` below anyway, we could invert this to become:

```suggestion
		if response.WasNotFound(fw.future.Response()) {
			err = nil
```